tests/integration/golang/aim/namespace/flows/app_test.go

```diff
@@ -0,0 +1,280 @@
+//go:build integration
+
+package flows
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/gofiber/fiber/v2"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/G-Research/fasttrackml/pkg/api/aim/request"
+	"github.com/G-Research/fasttrackml/pkg/api/aim/response"
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/common"
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/dao/models"
+	"github.com/G-Research/fasttrackml/tests/integration/golang/helpers"
+)
+
+type AppFlowTestSuite struct {
+	helpers.BaseTestSuite
+}
+
+func TestAppFlowTestSuite(t *testing.T) {
+	suite.Run(t, &AppFlowTestSuite{})
+}
+
+func (s *AppFlowTestSuite) TearDownTest() {
+	assert.Nil(s.T(), s.NamespaceFixtures.UnloadFixtures())
+}
+
+func (s *AppFlowTestSuite) Test_Ok() {
+	tests := []struct {
+		name           string
+		namespace1Code string
+		namespace2Code string
+	}{
+		{
+			name:           "TestCustomNamespaces",
+			namespace1Code: "namespace-1",
+			namespace2Code: "namespace-2",
+		},
+		{
+			name:           "TestExplicitDefaultAndCustomNamespaces",
+			namespace1Code: "default",
+			namespace2Code: "namespace-1",
+		},
+		{
+			name:           "TestImplicitDefaultAndCustomNamespaces",
+			namespace1Code: "",
+			namespace2Code: "namespace-1",
+		},
+	}
+
+	for _, tt := range tests {
+		s.T().Run(tt.name, func(T *testing.T) {
+			defer func() {
+				assert.Nil(s.T(), s.NamespaceFixtures.UnloadFixtures())
+			}()
+
+			// setup namespaces
+			for _, nsCode := range []string{"default", tt.namespace1Code, tt.namespace2Code} {
+				// ignore errors here since default exists on first run
+				//nolint:errcheck
+				_, _ = s.NamespaceFixtures.CreateNamespace(context.Background(), &models.Namespace{
+					Code:                nsCode,
+					DefaultExperimentID: common.GetPointer(int32(0)),
+				})
+			}
+
+			// run actual flow test over the test data.
+			s.testAppFlow(tt.namespace1Code, tt.namespace2Code)
+		})
+	}
+}
+
+func (s *AppFlowTestSuite) testAppFlow(
+	namespace1Code, namespace2Code string,
+) {
+	// create Apps
+	app1ID := s.createApp(namespace1Code, &request.CreateApp{
+		Type: "tf",
+		State: request.AppState{
+			"app-state-key": "app-state-value1",
+		},
+	})
+
+	app2ID := s.createApp(namespace2Code, &request.CreateApp{
+		Type: "mpi",
+		State: request.AppState{
+			"app-state-key": "app-state-value2",
+		},
+	})
+
+	// test `GET /apps` endpoint with namespace 1
+	resp := []response.App{}
+	require.Nil(
+		s.T(),
+		s.AIMClient().WithMethod(
+			http.MethodGet,
+		).WithNamespace(
+			namespace1Code,
+		).WithResponseType(
+			helpers.ResponseTypeJSON,
+		).WithResponse(
+			&resp,
+		).DoRequest(
+			"/apps",
+		),
+	)
+	// only app 1 should be present
+	assert.Equal(s.T(), 1, len(resp))
+	assert.Equal(s.T(), app1ID, resp[0].ID)
```

Comment:
why you didn't wrap it into method?

Reply:
extracted methods for the "positive" tests, but left error expectations in the main test method (awkward to share a method for this)